Lib/importlib/abc.py

```diff
@@ -382,23 +388,35 @@ def read_text(self, encoding=None):
     def is_dir(self) -> bool:
         """
         Return True if self is a dir
+
+        If the directory/file cannot be found, FileNotFoundError should be raised,
+        although this is not guaranteed for compatibility reasons.
         """
 
     @abc.abstractmethod
     def is_file(self) -> bool:
         """
         Return True if self is a file
+
+        If the directory/file cannot be found, FileNotFoundError should be raised,
```

Comment:
I'm not sure `FileNotFoundError` is correct here. Pathlib returns False when self is not a file.

```
>>> pathlib.Path('./does-not-exist').is_file()
False
```